new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSessionMigration.java

```diff
@@ -0,0 +1,462 @@
+    // create standard, reusable retry/backoff policy
+    withStandardRetryPolicy = with().pollDelay(2, SECONDS)
+      .and().with().pollInterval(10, SECONDS)
+      .atMost(5, MINUTES).await();
+
+    // get a unique operator namespace
+    logger.info("Get a unique namespace for operator");
+    assertNotNull(namespaces.get(0), "Namespace list is null");
+    opNamespace = namespaces.get(0);
+
+    // get a unique domain namespace
+    logger.info("Get a unique namespace for WebLogic domain");
+    assertNotNull(namespaces.get(1), "Namespace list is null");
+    domainNamespace = namespaces.get(1);
+
+    // install and verify operator
+    installAndVerifyOperator(opNamespace, domainNamespace);
+
+    // create and verify WebLogic domain image using model in image with model files
+    String imageName = createAndVerifyDomainImage();
+
+    // create and verify one cluster domain
+    logger.info("Create domain and verify that it's running");
+    createAndVerifyDomain(imageName);
+
+    // map to save HTTP response data
+    httpAttrMap = new HashMap<String, String>();
+    httpAttrMap.put("sessioncreatetime", "(.*)sessioncreatetime>(.*)</sessioncreatetime(.*)");
+    httpAttrMap.put("sessionid", "(.*)sessionid>(.*)</sessionid(.*)");
+    httpAttrMap.put("primary", "(.*)primary>(.*)</primary(.*)");
+    httpAttrMap.put("secondary", "(.*)secondary>(.*)</secondary(.*)");
+    httpAttrMap.put("count", "(.*)countattribute>(.*)</countattribute(.*)");
+  }
+
+  @AfterAll
+  void tearDown() {
+  }
+
+  /**
+   * The test sends a HTTP request to set http session state(count number), get the primary and secondary server name,
+   * session create time and session state and from the util method and save HTTP session info,
+   * then stop the primary server by changing ServerStartPolicy to NEVER and patching domain.
+   * Send another HTTP request to get http session state (count number), primary server and
+   * session create time. Verify that a new primary server is selected and HTTP session state is migrated.
+   */
+  @Test
+  @DisplayName("Stop the primary server, verify that a new primary server is picked and HTTP session state is migrated")
+  @Slow
+  @MustNotRunInParallel
+  public void testSessionMigration() {
+    final String primaryServerAttr = "primary";
+    final String secondaryServerAttr = "secondary";
+    final String sessionCreateTimeAttr = "sessioncreatetime";
+    final String countAttr = "count";
+    final String webServiceSetUrl = SESSMIGR_APP_WAR_NAME + "/?setCounter=" + SESSION_STATE;
+    final String webServiceGetUrl = SESSMIGR_APP_WAR_NAME + "/?getCounter";
+    String serverName = managedServerPrefix + "1";
+
+    // send a HTTP request to set http session state(count number) and save HTTP session info
+    // before shutting down the primary server
+    Map<String, String> httpDataInfo =
+        getServerAndSessionInfoAndVerify(serverName, webServiceSetUrl, " -D ");
```

Comment:
change method name to sendRequestAndGetSessionInfo() or similar to indicate what the method is doing

Reply:
changed